### PR TITLE
Pass sourceId to toggleContentShare

### PR DESCRIPTION
**Pull request by jing-chen1**
**Issue #:** [Allow to pass sourceId to toggleContentShare](https://github.com/aws/amazon-chime-sdk-component-library-react/issues/292)

**Description of changes:**
* Pass the `sourceId` as an optional parameter to the `toggleContentShare` function, so the builder can use it to start content share
* Add a new page, a "How-tos" section in the story book to show builders how to use the feature with some sample code. We can keep adding some particular use cases in this page.
* Also add a line in Readme, pointing to the "How-tos"

**Testing**
1. Have you successfully run `npm run build:release` locally? Yes

2. How did you test these changes? On my classroom demo, click the "content share" button in the control bar, and choose a screen, click "share" button. Then it starts content sharing.

3. If you made changes to the component library, have you provided corresponding documentation changes? Yes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +15,8 @@ The Amazon Chime React Component Library can be added to your web application co
 
 Documentation on these components and how to use them can be found on [Amazon Chime SDK React Component Library Documentation](https://aws.github.io/amazon-chime-sdk-component-library-react/).
 
+If you have some particular use cases, please go to [How-to guides](https://aws.github.io/amazon-chime-sdk-component-library-react/?path=/story/how-tos--page)
```
Comment (devalevenkatesh) [inline README.md:18]: May be we just say "Please check use-cases in our [How-to guides](https://aws.github.io/amazon-chime-sdk-component-library-react/?path=/story/how-tos--page) for more information" ?

File: src/components/howTos.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<Meta title="How-tos" />
+
+# How-to guides
+
+There are some particular use cases in this page. You can refer to the usage and some examples below.
```
Comment (devalevenkatesh) [inline src/components/howTos.stories.mdx:5]: How to guide can have more examples more generally and not only particular right? So, should we say:
"Please check use-cases below for usage examples."

File: src/providers/ContentShareProvider/docs/useContentShareControls.stories.mdx [hunk 3]
```diff
@@ -14,7 +14,8 @@ The `useContentShareControls` hook returns the state and functionality around st
   // A function to toggle content share.
   // If the user is sharing, it will stop the content share.
   // If a user is not sharing, it will start the content share workflow.
-  toggleContentShare: () => void;
+  // You can also provide a sourceId to show a specific screen.
```
Comment (devalevenkatesh) [inline src/providers/ContentShareProvider/docs/useContentShareControls.stories.mdx:17]: Nit: "show" -> "share" ?

File: src/types/index.ts [hunk 4]
```diff
@@ -34,7 +34,7 @@ export type LocalVideoContextType = {
 
 export type ContentShareControlContextType = {
   paused: boolean;
-  toggleContentShare: () => Promise<void>;
+  toggleContentShare: (sourceId?: string) => void;
```
Comment (devalevenkatesh) [inline src/types/index.ts:37]: Changing the return type is a breaking change right? So can we keep the return type same? Also, in the usage example in the how-to, you are using `await toggleContentShare`.

Also, one query we pass an optional attribute to start with, so adding a required variable later would be a breaking change / hard later. I do not know how to handle this or this is fine. @ltrung @simmkyu can you advice here?

File: README.md (hunk 1 above)
Comment (jing-chen1) [reply README.md:18]: Sure, will update it

File: src/components/howTos.stories.mdx (hunk 2 above)
Comment (jing-chen1) [reply src/components/howTos.stories.mdx:5]: emm, I can remove the first sentence.

File: src/providers/ContentShareProvider/docs/useContentShareControls.stories.mdx (hunk 3 above)
Comment (jing-chen1) [reply src/providers/ContentShareProvider/docs/useContentShareControls.stories.mdx:17]: Sure!

File: src/components/howTos.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+## Enabling content share in Electron applications
+
+If you are building a Electron app, you would need to create your own screen picker, and pass the `sourceId` to `useContentShareControls` hook to start content share.
```
Comment (simmkyu) [inline src/components/howTos.stories.mdx:9]: How about the following to briefly explain why the custom picker is needed?
> In an Electron application, you should create a custom screen picker to retrieve the screen's source ID and pass it to the `useContentShareControls` hook to start sharing your screen.

File: src/components/howTos.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  useContentShareControls,
+} from 'amazon-chime-sdk-component-library-react';
+
+const App = () => {
+  const { toggleContentShare } = useContentShareControls();
+
+  return (
+    <MeetingProvider>
+      <YourScreenPicker
+        onClickShareButton={async (selectedSourceId: string) => {
```
Comment (simmkyu) [inline src/components/howTos.stories.mdx:24]: We have JS SDK and component examples in JavaScript. Rewrite them into JavaScript.

File: src/components/howTos.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+          await new Promise((resolve) => setTimeout(resolve, 200));
```
Comment (simmkyu) [inline src/components/howTos.stories.mdx:25]: Remove this line as it does not explain any content-share stuff

File: src/types/index.ts (hunk 4 above)
Comment (simmkyu) [reply src/types/index.ts:37]: Yeah, recommend keeping the return type.

**Review by simmkyu — CHANGES_REQUESTED** [review]

File: src/components/howTos.stories.mdx (hunk 5 above)
Comment (jing-chen1) [reply src/components/howTos.stories.mdx:9]: Sure, will update it

File: README.md [hunk 8]
```diff
@@ -14,6 +15,8 @@ The Amazon Chime React Component Library can be added to your web application co
 
 Documentation on these components and how to use them can be found on [Amazon Chime SDK React Component Library Documentation](https://aws.github.io/amazon-chime-sdk-component-library-react/).
 
+Please check use-cases in our [How-to guides](https://aws.github.io/amazon-chime-sdk-component-library-react/?path=/story/how-tos--page) for more information.
```
Comment (devalevenkatesh) [inline README.md:18]: I think if we are having only 1 guide so better to change "guides" -> "guide"?

File: src/components/howTos.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,35 @@
+<Meta title="How-tos" />
+
+# How-to guides
+
+You can refer to the usage and some examples below.
+
+## Enabling content share in Electron applications
+
+In an Electron application, you should create a custom screen picker to retrieve the screen's source ID and pass it to the useContentShareControls hook to start sharing your screen.
```
Comment (devalevenkatesh) [inline src/components/howTos.stories.mdx:9]: Actually we do not pass the source ID to `useContentShareControls` hook right? So, should we say something like below:

"In an Electron application, you have to create a custom screen picker to retrieve the screen's source ID and pass it to the `toggleContentShare` method from `useContentShareControls` hook to start sharing your screen."

**Review by devalevenkatesh — CHANGES_REQUESTED** [review]
Few more comments:
1. Could you please add a changelog entry for this?
2. Could you please remove the `.tgz ` file added here?
3. This PR does not update any `package.json` so no `package-lock.json` update required right?

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,12 +7,21 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.6.0] - 2021-06-07
+
+### Added
+
+- Add `sourceId` as an optional parameter in the `toggleContentShare` function to specify which screen to share .
```
Comment (devalevenkatesh) [inline CHANGELOG.md:14]: Nit: remove space before ".".

File: CHANGELOG.md [hunk 11]
```diff
@@ -7,12 +7,21 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.6.0] - 2021-06-07
```
Comment (devalevenkatesh) [inline CHANGELOG.md:10]: So, if no, unrelease section is present here we add it like below:

## [Unreleased]

### Fixed

### Added

### Changed

### Removed

While releasing the [release script](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/master/scripts/release.js#L103) updates this [Unreleased] to specific version and date, hence you need not add the version and date here.

File: src/components/howTos.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,35 @@
+<Meta title="How-tos" />
+
+# How-to guides
+
+You can refer to the usage and some examples below.
+
+## Enabling content share in Electron applications
+
+In an Electron application, you should create a custom screen picker to retrieve the screen's source ID and pass it to the toggleContentShare method from useContentShareControls hook to start sharing your screen.
```
Comment (devalevenkatesh) [inline src/components/howTos.stories.mdx:9]: Could please backtick toggleContentShare and useContentShareControls here?

**Review by devalevenkatesh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 11 above)
Comment (jing-chen1) [reply CHANGELOG.md:10]: Thanks for the explanation, will update it

**Review by devalevenkatesh — APPROVED** [review]

**Review by simmkyu — APPROVED** [review]

**Pull request merged**